library/ssl_client.c

```diff
@@ -1015,14 +1015,24 @@ int mbedtls_ssl_write_client_hello( mbedtls_ssl_context *ssl )
         ssl->out_msglen = msg_len + 4;
         mbedtls_ssl_send_flight_completed( ssl );
 
+        /*
+         * The two functions below may try to send data on the network and
+         * can return with the MBEDTLS_ERR_SSL_WANT_READ error code when they
+         * fail to do so and the transmission has to be retried later. In that
+         * case as in fatal error cases, we return immediatly. But we must have
```

Comment:
Typo: immediatEly

(Side note: this (what the comment explains) is the kind of thing I find tricky when reasoning about TLS code. I wonder if long-term there's something we could to do make it less tricky.)

Reply:
Typo fixed. Otherwise, in TLS 1.3, there is now only one point where we send data and we will try to stick to that.